src/compiler/emitter.ts

```diff
@@ -2386,8 +2374,7 @@ module ts {
                             i++;
                         }
                         write("[");
-                        emitList(elements, pos, i - pos, /*multiLine*/ (node.flags & NodeFlags.MultiLine) !== 0,
-                            /*trailingComma*/ elements.hasTrailingComma);
+                        emitList(elements, pos, i - pos, multiLine, trailingComma);

```

Comment:
You're right. I think I'll just get rid of the trailingComma parameter. Given we're substantially rewriting the original code here there's really no point in preserving this detail.